common-content/en/module/logic/search/index.md

```diff
@@ -0,0 +1,119 @@
+
+{{<note title="A long awaited party" type="activity">}}
+**The King** has stocked his wine cellar with one thousand bottles of wine for his birthday party tomorrow. However, the King's evil brother, the Duke, has poisoned one of the bottles! The poison is lethal and will kill anyone who takes even one sip within one hour. You, the Butler, must find the poisoned bottle and save the party! But how?
+{{</note>}}
+
+> #### Conditions:
+>
+> 🍾 You have **1000** wine bottles, one of which is poisoned.  
+> 🐀 You have **500** rats you can use to test the wine.  
+> ⌛ The poison is lethal and will kill within **1** hour.
+
+{{<tabs name="boolean">}}
+===[[Write down your solution]]===
+
+In your notebook, write down your solution to this problem. Here are the constraints again:
+
+🍾 You have **1000** wine bottles, one of which is poisoned.  
+🐀 You have **500** rats you can use to test the wine.  
+⌛ The poison is lethal and will kill within **1** hour.
+
+===[[🍾 🍾 🐀 Look at a solution]]===
+
+### 🍾 🍾 🐀
+
+One way to think of this is to reduce this problem to its simplest case. If you have 2 bottles and 1 rat, you can test the wine by giving the rat a sip from one bottle. Wait one hour. If the rat is alive, the bottle is safe. If the rat is dead, the bottle is poisoned.
+
+With 500 rats, it's similar, but not the same. Divide the 1000 bottles into 500 groups of 2. In round 1, give each rat a sip from one bottle in each group. Wait one hour. If the rat is alive, the bottle is safe. If the rat is dead, you found the poisoned bottle. If all rats are alive, the poisoned bottle is in the 500 bottles that were not tested. You still have 500 rats, so you can have each rat test the other bottle in their group.
+
+This solution will take a maximum of 2 hours.
+
+{{</tabs>}}
+
+> #### New condition
+>
+> You now have just **10** rats to test the wine.
+
+{{<tabs name="decimation">}}
+===[[Write down your solution]]===
+
+In your notebook, write down your solution to this problem. Here are the constraints again:
+
+🍾 You have **1000** wine bottles, one of which is poisoned.  
+🐀 You have **10** rats you can use to test the wine.  
+⌛ The poison is lethal and will kill within **1** hour.
+
+===[[Look at a solution]]===
+
+With just 10 rats, we need a way to make our problem simpler. Think of it as a search pattern, narrowing down our suspects with each round of testing.
+
+**Round 1:** Let's divide our 1000 bottles into 10 groups of 100. Each rat tests one group. An hour later, with one dead rat, we have narrowed our search to 100 bottles. We've saved 900 bottles of wine and we still have 9 rats to help us.
+
+**Round 2** Let's do it again: take our suspicious 100 bottles and create ten groups of ten. Our doomed rat perishes again, and after another hour, we've found the right group of ten. We now have 8 rats.
```

Comment:
```suggestion
**Round 2** Let's do it again: take our suspicious 100 bottles and create ten groups of ten. Our nine rats each drink from one whole group (and one group goes un-drunk). After another hour, either a rat dies and we're down to ten bottles to investigate, or no rat died and we know the un-drunk group contains the poison. We now have at least 8 rats.
```

Reply:
I will take these changes but rewrite because these sentences are too convoluted. But I take the point